state_version_integration_test.go

```diff
@@ -318,26 +318,39 @@ func TestStateVersionsRead(t *testing.T) {
 	t.Cleanup(svTestCleanup)
 
 	t.Run("when the state version exists", func(t *testing.T) {
-		sv, err := client.StateVersions.Read(ctx, svTest.ID)
-		require.NoError(t, err)
-
-		// Don't compare the DownloadURL because it will be generated twice
-		// in this test - once at creation of the configuration version, and
-		// again during the GET.
-		svTest.DownloadURL, sv.DownloadURL = "", ""
-
-		// outputs are populated only once the state has been parsed by TFC
-		// which can cause the tests to fail if it doesn't happen fast enough.
-		svTest.Outputs = nil
-		sv.Outputs = nil
-
-		assert.Equal(t, svTest, sv)
-		assert.NotEmpty(t, svTest, svTest.ResourcesProcessed)
-		assert.NotEmpty(t, svTest, svTest.StateVersion)
-		assert.NotEmpty(t, svTest, svTest.TerraformVersion)
-		assert.NotEmpty(t, svTest, svTest.Modules)
-		assert.NotEmpty(t, svTest, svTest.Providers)
-		assert.NotEmpty(t, svTest, svTest.Resources)
+		if !svTest.ResourcesProcessed {
+			_, err := retry(func() (interface{}, error) {
+				svTest, err := client.StateVersions.Read(ctx, svTest.ID)
+				require.NoError(t, err)
+
+				if !svTest.ResourcesProcessed {
+					return nil, fmt.Errorf("resources not processed %s", err)
+				} else {
```

Comment:
I think it would be clearer to move this entire block outside of and right after retry (line 352 in this version)

That way, the retry can concentrate on awaiting ResourcesProcessed and then the test can proceed

The problem here is that if resources are processed on the very first attempt, the assertions are not run at all.

Reply:
The problem I ran into here, was accessing `svTest.ResourcesProcessed` when it becomes `true`. It will reference the global instance that is `false` rather than the updated value.

Reply:
Any suggestions?

Reply:
You can just assume it is true unless retry returns a non-nil error, because retry always returns an error if it remains false